gateway/api/connections/connections.go

```diff
@@ -680,3 +693,379 @@ func GetDatabaseSchemas(c *gin.Context) {
+				response.Schemas = append(response.Schemas, struct {
+					Name   string   `json:"name"`
+					Tables []string `json:"tables"`
+				}{
+					Name:   schemaName,
+					Tables: tables,
+				})
+			}
+		}
+
+		c.JSON(http.StatusOK, response)
+
+	case <-timeoutCtx.Done():
+		client.Close()
+		log.Infof("runexec timeout (30s), it will return async")
+		c.JSON(http.StatusBadRequest, gin.H{"message": "Request timed out"})
+	}
+}
+
+// GetTableColumns returns the columns of a specific table
+//
+//	@Summary		Get Table Columns
+//	@Description	Get columns from a specific table
+//	@Tags			Connections
+//	@Produce		json
+//	@Param			nameOrID	path		string	true	"Name or UUID of the connection"
+//	@Param			database	query		string	true	"Name of the database"
+//	@Param			table		query		string	true	"Name of the table"
+//	@Param			schema		query		string	true	"Name of the schema"
+//	@Success		200			{object}	ColumnsResponse
+//	@Failure		400,404,500	{object}	openapi.HTTPError
+//	@Router			/connections/{nameOrID}/columns [get]
+func GetTableColumns(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+	connNameOrID := c.Param("nameOrID")
+	dbName := c.Query("database")
+	tableName := c.Query("table")
+	schemaName := c.Query("schema")
+
+	if tableName == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "table parameter is required"})
+		return
+	}
+
+	conn, err := FetchByName(ctx, connNameOrID)
+	if err != nil {
+		log.Errorf("failed fetching connection, err=%v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+		return
+	}
+
+	if conn == nil {
+		c.JSON(http.StatusNotFound, gin.H{"message": "connection not found"})
+		return
+	}
+
+	if conn.Type != "database" {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "connection is not a database type"})
+		return
+	}
+
+	currentConnectionType := pb.ToConnectionType(conn.Type, conn.SubType.String)
+
+	if dbName == "" {
+		connEnvs := conn.Envs
+		switch currentConnectionType {
+		case pb.ConnectionTypePostgres,
+			pb.ConnectionTypeMSSQL,
+			pb.ConnectionTypeMySQL:
+			dbName = getEnvValue(connEnvs, "envvar:DB")
+		case pb.ConnectionTypeMongoDB:
+			if connStr := connEnvs["envvar:CONNECTION_STRING"]; connStr != "" {
+				dbName = getMongoDBFromConnectionString(connStr)
+			}
+		case pb.ConnectionTypeOracleDB:
+			dbName = getEnvValue(connEnvs, "envvar:SID")
+		}
+
+		if dbName == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"message": "database name is required but not found in connection or query parameter"})
+			return
+		}
+	}
+
+	// Validate database name to prevent SQL injection
+	if err := validateDatabaseName(dbName); err != nil {
+		c.JSON(http.StatusUnprocessableEntity, gin.H{"message": err.Error()})
+		return
+	}
+
+	// If the schema is not provided, use 'public' for PostgreSQL and the same database for other databases
+	if schemaName == "" {
+		if currentConnectionType == pb.ConnectionTypePostgres {
+			schemaName = "public"
+		} else {
+			schemaName = dbName
+		}
```

Comment:
Try to avoid using else when possible. The example below is more clear

```go
schemaName = dbName
if currentConnectionType == pb.ConnectionTypePostgres {
  schemaName = "public"
}
```